File: app/src/main/java/org/zeroxlab/momodict/ui/FragmentListener.java
package org.zeroxlab.momodict.ui;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

/**
 * This is a listener to for any message from fragments
 */
public interface FragmentListener {
    enum TYPE {
        START_ACTIVITY
    }

    void onNotified(@Nullable Fragment from,
                    @NonNull TYPE type,
                    @Nullable Object payload);
}

File: app/src/main/java/org/zeroxlab/momodict/ui/FragmentListener.kt
package org.zeroxlab.momodict.ui

import android.support.v4.app.Fragment

/**
 * This is a listener to for any message from fragments
 */
interface FragmentListener {
    enum class TYPE {
        START_ACTIVITY,
        UPDATE_TITLE,
        POP_FRAGMENT,
        VIEW_ACTION
    }

    fun onNotified(from: Fragment?,
                   type: TYPE,
                   payload: Any?)
}
